### PR TITLE
Pass mutated properties through to 'bucket.create' API call.

### PR DESCRIPTION
**Pull request by tseaver**
Fixes: #1575.

### CODE REVIEW

File: gcloud/storage/bucket.py [hunk 1]
```diff
@@ -167,9 +167,12 @@ def create(self, client=None):
         """
         client = self._require_client(client)
         query_params = {'project': client.project}
+        properties = dict(
+            (key, self._properties[key]) for key in self._changes)
```
Comment (dhermes) [inline gcloud/storage/bucket.py]: It's just like a dict-comprehension and it works in Python 2.6. Nice.

File: gcloud/storage/bucket.py [hunk 2]
```diff
@@ -167,9 +167,12 @@ def create(self, client=None):
...
+        properties['name'] = self.name
```
Comment (dhermes) [inline gcloud/storage/bucket.py]: What is `name` was already in there? Or is it impossible to add `name` to `_changes`?

Comment (tseaver) [inline gcloud/storage/bucket.py]: `name` is not a settable property.

Comment (tseaver) [inline gcloud/storage/bucket.py]: and therefore cannot be in `self._changes` (I forgot to add).

File: gcloud/storage/bucket.py (hunk 1 above)
Comment (tseaver) [inline gcloud/storage/bucket.py]: Kids these days and their sweet-tooth obsessions. :)

Comment (dhermes) [inline gcloud/storage/bucket.py]: Ha

**Comment by dhermes** [comment]
LGTM

**Pull request merged**